cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/validation/TruststoreValidation.java

```diff
@@ -0,0 +1,22 @@
+package org.apache.cassandra.spark.validation;
+
+import org.apache.cassandra.secrets.SecretsProvider;
+
+public class TruststoreValidation implements StartupValidation
+{
+    private final SecretsProvider secrets;
+
+    public TruststoreValidation(SecretsProvider secrets)
+    {
+        this.secrets = secrets;
+    }
+
+    @Override
+    public void validate()
+    {
+        if (!secrets.hasTrustStoreSecrets())
```

Comment:
hmm, I'm not sure if this is sufficient to validate truststore

Reply:
What else should happen here?

Reply:
well, a truststore should be optional, if not provided the JVM will pick up the default truststore. But if provided, we need to make sure that the keystore is valid with the provided truststore. Errors will look like "certification path can't be established" or something along those lines

Reply:
Added everything we've discussed in a separate commit, could you take another look?